### PR TITLE
fix: nullable in schema object must have a type sibling

### PR DESCRIPTION
**Pull request by AntonKozachuk**
## What/Why/How?
We need to let the user know where incorrect use of nullable has happened.

## Reference
#925 

## Testing
Locally.

## Screenshots (optional)
N/a

## Check yourself

- [ ] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

**Comment by AntonKozachuk** [comment]
@adamaltman with this fix the Rebilly spec we have added to CLI and used in benchmarks is failing: https://github.com/Redocly/redocly-cli/actions/runs/3478505767/jobs/5816135478
Is this expected?

File: packages/core/src/rules/common/spec.ts [hunk 1]
```diff
@@ -158,6 +158,13 @@ export const OasSpec: Oas3Rule | Oas2Rule = () => {
             });
           }
         }
+
+        if (propName === 'nullable' && !node.type) {
+          report({
+            message: '`Type` keyword should be explicitly defined when `nullable` keyword is used.',
```
Comment (RomanHotsiy) [inline packages/core/src/rules/common/spec.ts:164]:
```suggestion
            message: '`type` keyword should be explicitly defined when `nullable` keyword is used.',
```

**Review by RomanHotsiy — APPROVED** [review]
LGTM

File: __tests__/lint/no-invalid-media-type-examples-invalid-schema/snapshot.js [hunk 2]
```diff
@@ -3,7 +3,21 @@
 exports[`E2E lint no-invalid-media-type-examples-invalid-schema 1`] = `
 
 validating /openapi.yaml...
-[1] openapi.yaml:28:17 at #/paths/~1test/get/responses/202/content/application~1json/schema
+[1] openapi.yaml:16:17 at #/components/schemas/Test/nullable
+
+\`Type\` keyword should be explicitly defined when \`nullable\` keyword is used.
```
Comment (adamaltman) [inline __tests__/lint/no-invalid-media-type-examples-invalid-schema/snapshot.js:8]:
```suggestion
The \`type\` field must be defined when the \`nullable\` field is used.
```
Change to this error message everywhere.

**Review by tatomyr — APPROVED** [review]

**Pull request merged**